src/services/matchZendeskTicket.spec.ts

```diff
@@ -0,0 +1,228 @@
+            value: 'event_id'
+          },
+          {
+            id: '6',
+            value: null
+          },
+          {
+            id: '7',
+            value: 'passport_number'
+          },
+          {
+            id: '8',
+            value: null
+          },
+          {
+            id: '9',
+            value: null
+          }
+        ]
+      })
+    )
+    mockGetZendeskUser.mockImplementation(() =>
+      Promise.resolve({
+        email: 'myuser@test.gov.uk',
+        name: 'my name'
+      })
+    )
+  }
+
+  const givenZendeskTicketDoesNotMatch = () => {
+    mockGetZendeskTicket.mockImplementation(() =>
+      Promise.resolve({
+        id: '1234',
+        requester_id: '123',
+        custom_fields: [
+          {
+            id: '1',
+            value: '123'
+          },
+          {
+            id: '2',
+            value: '2019-10-12'
+          },
+          {
+            id: '3',
+            value: '2022-10-12'
+          },
+          {
+            id: '4',
+            value: '456 678'
+          },
+          {
+            id: '5',
+            value: 'event_id'
+          },
+          {
+            id: '6',
+            value: '123'
+          },
+          {
+            id: '7',
+            value: 'passport_number name'
+          },
+          {
+            id: '8',
+            value: null
+          },
+          {
+            id: '9',
+            value: null
+          }
+        ]
+      })
+    )
+
+    mockGetZendeskUser.mockImplementation(() =>
+      Promise.resolve({
+        email: 'notmyuser@test.gov.uk',
+        name: 'not my name'
+      })
+    )
+  }
+
+  const givenCustomFieldNotFound = () => {
+    mockGetZendeskTicket.mockImplementation(() =>
+      Promise.resolve({
+        id: ZENDESK_TICKET_ID,
+        requester_id: '123',
+        custom_fields: [
+          {
+            id: '2',
+            value: TEST_DATE_FROM
+          },
+          {
+            id: '3',
+            value: TEST_DATE_TO
+          },
+          {
+            id: '4',
+            value: '123 456'
+          },
+          {
+            id: '5',
+            value: 'event_id'
+          },
+          {
+            id: '6',
+            value: null
+          },
+          {
+            id: '7',
+            value: 'passport_number'
+          },
+          {
+            id: '8',
+            value: null
+          },
+          {
+            id: '9',
+            value: null
+          }
+        ]
+      })
+    )
+    mockGetZendeskUser.mockImplementation(() =>
+      Promise.resolve({
+        email: 'myuser@test.gov.uk',
+        name: 'my name'
+      })
+    )
+  }
+
+  test('ticket and request match', async () => {
+    givenZendeskTicketMatches()
+
+    const matchResult = await matchZendeskTicket(testDataRequest)
+
+    expect(matchResult).toEqual(true)
+  })
+
+  test('ticket and request do not match', async () => {
+    givenZendeskTicketDoesNotMatch()
```

Comment:
How have you selected which case to check for here? Given that we have code to check each field, ideally you'd want to test each one. You can probably find a parameterised way to do this to avoid repetition, but for full test coverage I think you'll need to do that.